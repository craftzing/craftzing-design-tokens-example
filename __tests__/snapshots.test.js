import { readFile } from 'fs/promises';
import path from 'path';
import { expect, describe, it } from 'vitest';

const BUILD_PATH = './dist';
const tokenTypes = ['figma-variables', 'token-studio'];
const formats = ['css', 'scss', 'js', 'ts'];

describe('Token Build Snapshot Tests', () => {
  it.each(
    tokenTypes.flatMap((tokenType) =>
      formats.map((format) => ({
        tokenType,
        format,
        fileExtension: format === 'ts' ? 'd.ts' : format,
      }))
    )
  )(
    'should match the snapshot for $tokenType $format tokens',
    async ({ tokenType, format, fileExtension }) => {
      const tokenFilePath = path.join(
        BUILD_PATH,
        tokenType,
        format,
        `tokens.${fileExtension}`
      );
      const tokenFileContent = await readFile(tokenFilePath, 'utf-8');
      expect(tokenFileContent).toMatchSnapshot();
    }
  );
});
