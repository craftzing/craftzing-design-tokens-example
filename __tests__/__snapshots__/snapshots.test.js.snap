// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Token Build Snapshot Tests > should match the snapshot for 'figma-variables' 'css' tokens 1`] = `
"/**
 * Do not edit directly, this file was auto-generated.
 */

:root {
  --primary: #1a297c;
  --secondary: #ef268b;
  --white: #ffffff;
  --black: #000000;
  --button-background: #1a297c;
  --button-foreground: #ffffff;
}
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'figma-variables' 'js' tokens 1`] = `
"/**
 * Do not edit directly, this file was auto-generated.
 */

export const primary = "#1a297c";
export const secondary = "#ef268b";
export const white = "#ffffff";
export const black = "#000000";
export const buttonBackground = "#1a297c";
export const buttonForeground = "#ffffff";
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'figma-variables' 'scss' tokens 1`] = `
"
// Do not edit directly, this file was auto-generated.

$primary: #1a297c;
$secondary: #ef268b;
$white: #ffffff;
$black: #000000;
$button-background: #1a297c;
$button-foreground: #ffffff;
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'figma-variables' 'ts' tokens 1`] = `
"/**
 * Do not edit directly, this file was auto-generated.
 */

export const primary: "#1a297c";
export const secondary: "#ef268b";
export const white: "#ffffff";
export const black: "#000000";
export const buttonBackground: "#1a297c";
export const buttonForeground: "#ffffff";
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'token-studio' 'css' tokens 1`] = `
"/**
 * Do not edit directly, this file was auto-generated.
 */

:root {
  --foundations-colors-primary: #1a297c;
  --foundations-colors-secondary: #ef268b;
  --foundations-colors-white: #ffffff;
  --foundations-colors-black: #000000;
  --typography-font-sizes-small: 0.75rem;
  --typography-font-sizes-medium: 1rem;
  --typography-font-sizes-large: 1.5rem;
  --components-button-background: #1a297c;
  --components-button-foreground: #ffffff;
  --components-button-typography: 400 16px "Roboto", system-ui;
}
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'token-studio' 'js' tokens 1`] = `
"/**
 * Do not edit directly, this file was auto-generated.
 */

export const foundationsColorsPrimary = "#1a297c";
export const foundationsColorsSecondary = "#ef268b";
export const foundationsColorsWhite = "#ffffff";
export const foundationsColorsBlack = "#000000";
export const typographyFontSizesSmall = "12px";
export const typographyFontSizesMedium = "16px";
export const typographyFontSizesLarge = "24px";
export const componentsButtonBackground = "#1a297c";
export const componentsButtonForeground = "#ffffff";
export const componentsButtonTypography = {
  fontFamily: '"Roboto", system-ui',
  fontWeight: "400",
};
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'token-studio' 'scss' tokens 1`] = `
"
// Do not edit directly, this file was auto-generated.

$foundations-colors-primary: #1a297c;
$foundations-colors-secondary: #ef268b;
$foundations-colors-white: #ffffff;
$foundations-colors-black: #000000;
$typography-font-sizes-small: 0.75rem;
$typography-font-sizes-medium: 1rem;
$typography-font-sizes-large: 1.5rem;
$components-button-background: #1a297c;
$components-button-foreground: #ffffff;
$components-button-typography: 400 16px "Roboto", system-ui;
"
`;

exports[`Token Build Snapshot Tests > should match the snapshot for 'token-studio' 'ts' tokens 1`] = `
"/**
 * Do not edit directly, this file was auto-generated.
 */

export const foundationsColorsPrimary: "#1a297c";
export const foundationsColorsSecondary: "#ef268b";
export const foundationsColorsWhite: "#ffffff";
export const foundationsColorsBlack: "#000000";
export const typographyFontSizesSmall: "12px";
export const typographyFontSizesMedium: "16px";
export const typographyFontSizesLarge: "24px";
export const componentsButtonBackground: "#1a297c";
export const componentsButtonForeground: "#ffffff";
export const componentsButtonTypography: {
  fontFamily: string;
  fontWeight: string;
};
"
`;
